elpis/endpoints/model.py

```diff
@@ -141,6 +150,77 @@ def results():
     return jsonify({"status": 200, "data": data})
 
 
+@bp.route("/download", methods=["GET", "POST"])
+def download():
+    """Downloads the model files to the frontend"""
+    model: HFTModel = app.config["CURRENT_MODEL"]
+    if model is None:
+        logger.error("No current model exists")
+        return jsonify(MISSING_MODEL_RESPONSE)
+
+    zipped_model_path = Path("/tmp", "model.zip")
+    # zipped_model_path.mkdir(parents=True, exist_ok=True)
+    logger.info(f"Creating zipped model at path: {zipped_model_path}")
+    shutil.make_archive(
+        str(zipped_model_path.parent / zipped_model_path.stem), "zip", model.path / MODEL_PATH
+    )
+    logger.info((f"Zipped model created at path: {zipped_model_path}"))
+    return send_file(zipped_model_path, as_attachment=True, cache_timeout=0)
+
+
+@bp.route("/upload", methods=["POST"])
+def upload():
+    logger.info("Upload endpoint started")
+    engine = ENGINES["hft"]
+    interface: Interface = app.config["INTERFACE"]
+    interface.set_engine(engine)
+
+    # Save files to model directory
+    zip_file = request.files.getlist("file")[0]
+    filename = secure_filename(str(zip_file.filename))
+
+    if filename == "" or Path(filename).suffix != ".zip":
+        return jsonify({"status": 500, "error": "Invalid filename or not a zip-file"})
+
+    try:
+        model: HFTModel = interface.new_model(Path(filename).stem)
+        logger.info(f"New model created {model.name} {model.hash}")
+    except InterfaceError as e:
+        return jsonify({"status": 500, "error": e.human_message})
+    app.config["CURRENT_MODEL"] = model
+
+    zip_path = model.output_dir / filename
+    logger.info(f"Saving the zipped model at {zip_path}")
+    zip_path.parent.mkdir(parents=True, exist_ok=True)
+
+    zip_file.save(zip_path)
+    shutil.unpack_archive(zip_path, model.output_dir)
+    os.remove(zip_path)
+    logger.info(f"Zipped model unpacked and deleted")
+
+    # Attempts to unpack a zip file if when unzipped, it resolves to a single directory.
+    folder_path = zip_path.parent / zip_path.stem
+    if folder_path.exists and folder_path.is_dir():
+        for file in os.listdir(folder_path):
+            (folder_path / file).rename(folder_path.parent / file)
+        folder_path.rmdir()
+
+    # Update model state
+    model.status = FINISHED
+    model_list = [
+        {
+            "name": model["name"],
+            "engine_name": model["engine_name"],
+            "status": model["status"],
+            "dataset_name": "UPLOADED_MODEL",
```

Comment:
I know this was apparently needed to make things not error on the frontend, but I'm thinking if necessary, in a future PR, we should differentiate between model types (trained vs uploaded) and then we won't have to make dummy values like this or weird workarounds for handling them.